drift/core/psestimation.py

```diff
@@ -638,8 +665,8 @@ def q_estimator(self, mi, vec1, vec2=None, noise=False):
 
             for li in range(lside):
 
-                lxvec = x2[:, 0, li]
-                lyvec = y2[:, 0, li]
+                lxvec = x2[:, li]
+                lyvec = y2[:, li]
```

Comment:
Tracing the function calls back, it looks like `x2` and `y2` will still have shape `(nfreq, npol, lmax+1, ...)`: `project_vector_kl_to_sky` takes its output from `project_vector_svd_to_sky`. Fixing the `temponly=True` option of `project_vector_svd_to_sky` would solve this, but as I mentioned in another comment, we'd want to be sure that that fix doesn't break anything else...

Reply:
`project_vector_kl_to_sky` actually slices x2 for `temponly`, see comment in line 623. so if we keep things as they are - everything works.

Reply:
Gotcha